drivers/video/vnc/vnc_updater.c

```diff
@@ -181,23 +182,24 @@ static FAR struct vnc_fbupdate_s *
 vnc_alloc_update(FAR struct vnc_session_s *session)
 {
   FAR struct vnc_fbupdate_s *update;
+  irqstate_t flags;
 
   /* Reserve one element from the free list.  Lock the scheduler to assure
    * that the sq_remfirst() and the successful return from nxsem_wait are
    * atomic.  Of course, the scheduler will be unlocked while we wait.
    */
 
-  sched_lock();
   vnc_sem_debug(session, "Before alloc", 0);
+  flags = enter_critical_section();
 
   nxsem_wait_uninterruptible(&session->freesem);
 
   /* It is reserved.. go get it */
 
   update = (FAR struct vnc_fbupdate_s *)sq_remfirst(&session->updfree);
```

Comment:
Can `session->updfree` be accessed from different cores?

Reply:
Yes, since it's accessed by two thread and then possible on SMP.